stko/optimizers/rdkit.py

```diff
@@ -0,0 +1,179 @@
+"""
+RDKit Optimizers
+================
+
+#. :class:`.MMFF`
+#. :class:`.UFF`
+#. :class:`.ETKDG`
+
+Wrappers for optimizers within the :mod:`rdkit` code.
+
+.. code-block:: python
+
+    import stk
+    import stko
+
+    # Optimizers work on stk.Molecule.
+    polymer = stk.ConstructedMolecule(
+        building_blocks=[mol],
+        topology_graph=stk.polymer.Linear('A', [0], n=3)
+    )
+    etkdg = stk.ETKDG()
+    mol = etkdg.optimize(polymer)
+
+"""
+
+import logging
+import rdkit.Chem.AllChem as rdkit
+
+from .optimizers import Optimizer
+
+
+logger = logging.getLogger(__name__)
+
+
+class MMFF(Optimizer):
+    """
+    Use the MMFF force field to optimize molecules.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        import stk
+
+        mol = stk.BuildingBlock('NCCNCCN', ['amine'])
+        mmff = stk.MMFF()
+        mol = mmff.optimize(mol)
+
+    """
+
+    def optimize(self, mol):
+        """
+        Optimize `mol`.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to be optimized.
+
+        Returns
+        -------
+        mol : :class:`.Molecule`
+            The optimized molecule.
+
+        """
+
+        rdkit_mol = mol.to_rdkit_mol()
+        # Needs to be sanitized to get force field params.
+        rdkit.SanitizeMol(rdkit_mol)
+        rdkit.MMFFOptimizeMolecule(rdkit_mol)
+        mol = mol.with_position_matrix(
+            position_matrix=rdkit_mol.GetConformer().GetPositions()
+        )
+
+
+class UFF(Optimizer):
+    """
+    Use the UFF force field to optimize molecules.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        import stk
+
+        mol = stk.BuildingBlock('NCCNCCN', ['amine'])
+        uff = stk.UFF()
+        mol = uff.optimize(mol)
+
+    """
+
+    def optimize(self, mol):
+        """
+        Optimize `mol`.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The molecule to be optimized.
+
+        Returns
+        -------
+        mol : :class:`.Molecule`
+            The optimized molecule.
+
+        """
+
+        rdkit_mol = mol.to_rdkit_mol()
+        # Needs to be sanitized to get force field params.
+        rdkit.SanitizeMol(rdkit_mol)
+        rdkit.UFFOptimizeMolecule(rdkit_mol)
+        mol = mol.with_position_matrix(
+            position_matrix=rdkit_mol.GetConformer().GetPositions()
+        )
+
+        return mol
+
+
+class ETKDG(Optimizer):
+    """
+    Uses the ETKDG [#]_ v2 algorithm to find an optimized structure.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        import stk
+
+        mol = stk.BuildingBlock('NCCNCCN', ['amine'])
+        etkdg = stk.ETKDG()
+        mol = etkdg.optimize(mol)
+
+    References
+    ----------
+    .. [#] http://pubs.acs.org/doi/pdf/10.1021/acs.jcim.5b00654
+
+    """
+
+    def __init__(self, random_seed=12):
```

Comment:
Shouldn't the default be 0 for a random seed?

Reply:
No, random seed of 0 is not allowed I do not think. it is not crucial what the number is though, as long as its constant